home-application/src/main/java/com/softserveinc/ita/homeproject/application/config/EnumConverterProvider.java

```diff
@@ -0,0 +1,58 @@
+package com.softserveinc.ita.homeproject.application.config;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import javax.ws.rs.ext.Provider;
+
+import org.modelmapper.spi.ErrorMessage;
+
+@SuppressWarnings("unchecked")
+@Provider
+public class EnumConverterProvider implements ParamConverterProvider {
+
+    @Override
+    public <T> ParamConverter<T> getConverter(Class<T> rawType, Type genericType, Annotation[] annotations) {
+        if (!rawType.isEnum()) {
+            return null;
+        }
+
+        final Class<Enum<?>> type = (Class<Enum<?>>) rawType;
+        final Enum<?>[] constants = type.getEnumConstants();
+
+        return new ParamConverter<T>() {
+            @Override
+            public T fromString(String value) {
+                if (value == null || value.isEmpty()) {
+                    return null;
+                }
+
+                for (Enum<?> constant : constants) {
+                    if (constant.toString().equals(value)) {
+                        return (T) constant;
+                    }
+                }
+
+                throw new BadRequestException(
+                    getErrorResponse("Parameter type must be one of enum constants"));
+            }
+
+            @Override
+            public String toString(T value) {
+                return value.toString();
+            }
+
+            protected Response getErrorResponse(String message) {
```

Comment:
I think that forming error response here isn't correct thus we have a more precise place for such things - ExceptionMappers.